src/components/MoneyRequestHeader.tsx

```diff
@@ -1,87 +1,73 @@
 import {usePersonalDetails} from './OnyxProvider';
-import transactionPropTypes from './transactionPropTypes';
 
-const propTypes = {
-    /** The report currently being looked at */
-    report: iouReportPropTypes.isRequired,
-
-    /** The policy which the report is tied to */
-    policy: PropTypes.shape({
-        /** Name of the policy */
-        name: PropTypes.string,
-    }),
-
-    /* Onyx Props */
+type MoneyRequestHeaderOnyxProps = {
     /** Session info for the currently logged in user. */
-    session: PropTypes.shape({
-        /** Currently logged in user email */
-        email: PropTypes.string,
-    }),
+    session: OnyxEntry<Session>;
 
     /** The expense report or iou report (only will have a value if this is a transaction thread) */
-    parentReport: iouReportPropTypes,
-
-    /** The report action the transaction is tied to from the parent report */
-    parentReportAction: PropTypes.shape(reportActionPropTypes),
+    parentReport: OnyxEntry<Report>;
 
     /** All the data for the transaction */
-    transaction: transactionPropTypes,
+    transaction: OnyxEntry<Transaction>;
+
+    /** All report actions */
+    // eslint-disable-next-line react/no-unused-prop-types
+    parentReportActions: OnyxEntry<ReportActions>;
 };
 
-const defaultProps = {
-    session: {
-        email: null,
-    },
-    parentReport: {},
-    parentReportAction: {},
-    transaction: {},
-    policy: {},
+type MoneyRequestHeaderProps = MoneyRequestHeaderOnyxProps & {
+    /** The report currently being looked at */
+    report: Report;
+
+    /** The policy which the report is tied to */
+    policy: Policy;
+
+    /** The report action the transaction is tied to from the parent report */
+    parentReportAction: ReportAction & OriginalMessageIOU;
 };
 
-function MoneyRequestHeader({session, parentReport, report, parentReportAction, transaction, policy}) {
+function MoneyRequestHeader({session, parentReport, report, parentReportAction, transaction, policy}: MoneyRequestHeaderProps) {
     const personalDetails = usePersonalDetails() || CONST.EMPTY_OBJECT;
     const styles = useThemeStyles();
     const {translate} = useLocalize();
     const [isDeleteModalVisible, setIsDeleteModalVisible] = useState(false);
     const moneyRequestReport = parentReport;
-    const isSettled = ReportUtils.isSettled(moneyRequestReport.reportID);
+    const isSettled = ReportUtils.isSettled(moneyRequestReport?.reportID);
     const isApproved = ReportUtils.isReportApproved(moneyRequestReport);
     const {isSmallScreenWidth, windowWidth} = useWindowDimensions();
 
     // Only the requestor can take delete the request, admins can only edit it.
-    const isActionOwner = lodashGet(parentReportAction, 'actorAccountID') === lodashGet(session, 'accountID', null);
+    const isActionOwner = typeof parentReportAction.actorAccountID === 'number' && typeof session?.accountID === 'number' && parentReportAction.actorAccountID === session?.accountID;
```

Comment:
Thanks! This reads clean
Should we safely check `parentReportAction?.actorAccountID` or is it fine?

Reply:
I took a closer look at the screen that uses this component and there's the possibility that parentReportAction is undefined, so I applied you suggestion! Thanks for the heads up, good catch 😄